server/app/services/rpc/container_handler.rb

```diff
@@ -0,0 +1,126 @@
+require_relative 'fixnum_helper'
+require_relative 'container_info_mapper'
+
+module Rpc
+  class ContainerHandler
+    include Celluloid
+    include FixnumHelper
+    include Logging
+
+    def initialize(grid)
+      @grid = grid
+      @logs = []
+      @stats = []
+      @cached_container = nil
+      @db_session = ContainerLog.collection.session.with(
+        write: {
+          w: 0, fsync: false, j: false
+        }
+      )
+    end
+
+    # @param [Hash] data
+    def save(data)
+      info_mapper = ContainerInfoMapper.new(@grid)
+      info_mapper.from_agent(data)
+    end
+
+    # @param [Hash] data
+    def log(data)
+      container = cached_container(data['id'])
+      if container
+        if data['time']
+          created_at = Time.parse(data['time'])
+        else
+          created_at = Time.now.utc
+        end
+        @logs << {
+          grid_id: @grid.id,
+          host_node_id: container['host_node_id'],
+          grid_service_id: container['grid_service_id'],
+          container_id: container['_id'],
+          created_at: created_at,
+          name: container['name'],
+          type: data['type'],
+          data: data['data']
+        }
+        if @logs.size >= 5
+          flush_logs
+        end
+      end
+    end
+
+    def flush_logs
```

Comment:
Confirmed that the following experiment results in duplicated logs:

```patch
--- a/server/app/services/rpc/container_handler.rb
+++ b/server/app/services/rpc/container_handler.rb
@@ -52,6 +52,7 @@ module Rpc
 
     def flush_logs
       @db_session[:container_logs].insert(@logs)
+      sleep 5
       @logs.clear
     end
 ```

Just something to watch out for and consider when working on these actors...

The results look amusing when restarting redis :)

![2017-03-01-134656_665x2159_scrot](https://cloud.githubusercontent.com/assets/3950373/23458606/8f436484-fe85-11e6-9b6c-93c664b419b5.png)